lib/dash/content_protection.js

```diff
@@ -262,6 +262,194 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
+    const begin = byteOffset / 2;
+    const end = (byteLength + byteOffset) / 2;
+
+    // subarray end is exclusive
+    const recordValue = new Uint16Array(recordData).subarray(begin, end);
+
+    records.push({
+      type: type,
+      value: new Uint8Array(recordValue),
+    });
+
+    byteOffset += byteLength;
+  }
+
+  return records;
+};
+
+
+/**
+ * Parses an ArrayBuffer for PlayReady Objects.  The data
+ * should contain a 32-bit integer indicating the length of
+ * the PRO in bytes.  Following that, a 16-bit integer for
+ * the number of PlayReady Object Records in the PRO.  Lastly,
+ * a byte array of the PRO Records themselves.
+ *
+ * PlayReady Object format: https://goo.gl/W8yAN4
+ *
+ * @param {!ArrayBuffer} data
+ * @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+ * @private
+ */
+shaka.dash.ContentProtection.parseMsPro_ = function(data) {
+  let byteOffset = 0;
+  const view = new DataView(data);
+
+  // First 4 bytes is the PRO length (DWORD)
+  const byteLength = view.getUint32(byteOffset, true /* littleEndian */);
+  byteOffset += 4;
+
+  if (byteLength !== data.byteLength) {
+    // Malformed PRO
+    shaka.log.warning('PlayReady Object with invalid length encountered.');
+    return [];
+  }
+
+  // Skip PRO Record count (WORD)
+  byteOffset += 2;
+
+  // Rest of the data contains the PRO Records
+  const ContentProtection = shaka.dash.ContentProtection;
+  return ContentProtection.parseMsProRecords_(data, byteOffset);
+};
+
+
+/**
+ * PlayReady Header format: https://goo.gl/dBzxNA
+ *
+ * @param {!Element} xml
+ * @return {string}
+ * @private
+ */
+shaka.dash.ContentProtection.getLaurl_ = function(xml) {
+  // LA_URL element is optional and no more than one is
+  // allowed inside the DATA element. Only absolute URLs are allowed.
+  // If the LA_URL element exists, it must not be empty.
+  const laurlNode = xml.querySelector('DATA > LA_URL');
+  if (laurlNode) {
+    return laurlNode.textContent;
+  }
+
+  // Not found
+  return '';
+};
+
+
+/**
+ * Gets a PlayReady license URL from a content protection element
+ * containing a PlayReady Header Object
+ *
+ * @param {shaka.dash.ContentProtection.Element} element
+ * @return {string}
+ */
+shaka.dash.ContentProtection.getPlayReadyLicenseUrl = function(element) {
+  const proNode = shaka.util.XmlUtils.findChildNS(
+    element.node, 'urn:microsoft:playready', 'pro');
+
+  if (!proNode) {
+    return '';
+  }
+
+  const ContentProtection = shaka.dash.ContentProtection;
+  const PLAYREADY_RECORD_TYPES = ContentProtection.PLAYREADY_RECORD_TYPES;
+
+  const bytes = shaka.util.Uint8ArrayUtils.fromBase64(proNode.textContent);
+  const records = ContentProtection.parseMsPro_(bytes.buffer);
+  const record = records.filter((record) => {
+    return record.type === PLAYREADY_RECORD_TYPES.RIGHTS_MANAGEMENT;
+  })[0];
+
+  if (!record) {
+    return '';
+  }
+
+  const rootElement = shaka.util.XmlUtils.parseXml(
+    record.value.buffer, 'WRMHEADER');
```

Comment:
This won't work since it will parse from the beginning of the buffer, not the beginning of the record.  You should edit the `parseXml` to accept a `BufferSource` instead and just pass `record.value`.

Reply:
No longer using `parseXml` so I didn't edit it's signature.